src/Sylius/Bundle/ApiBundle/spec/Modifier/OrderAddressModifierSpec.php

```diff
@@ -45,11 +45,11 @@ function it_handles_addressing_an_order_without_provided_shipping_address(
         $order->setBillingAddress($billingAddress)->shouldBeCalled();
         $order->setShippingAddress(Argument::type(AddressInterface::class))->shouldBeCalled();
 
-        $stateMachineFactory->get($order, OrderCheckoutTransitions::GRAPH)->willReturn($stateMachine);
-        $stateMachine->can(OrderCheckoutTransitions::TRANSITION_ADDRESS)->willReturn(true);
-        $stateMachine->apply(OrderCheckoutTransitions::TRANSITION_ADDRESS)->shouldBeCalled();
+        $stateMachineFactory->get($order, AsynchronousOrderCheckoutTransitions::GRAPH)->willReturn($stateMachine);
+        $stateMachine->can(AsynchronousOrderCheckoutTransitions::TRANSITION_ADDRESS)->willReturn(true);
+        $stateMachine->apply(AsynchronousOrderCheckoutTransitions::TRANSITION_ADDRESS)->shouldBeCalled();
 
-        $this->modify($order, $billingAddress->getWrappedObject(), null, 'r2d2@droid.com');
```

Comment:
shouldnt we keep both functionalities? Why was this email removed if the test was working on default checkout?

Reply:
The email wes removed because the interface uses only 3 parameters. Look at the  definition XD

```php
interface OrderAddressModifierInterface
{
    public function modify(OrderInterface $order, AddressInterface $billingAddress, ?AddressInterface $shippingAddress = null): OrderInterface;
}
```